### PR TITLE
Fix withdraw from private key for sundai

### LINKED ISSUE #29: Withdraw from another address not working (+other issues)
Precoditions:
* User already has burner wallet with some pDAI/sunDAI
* QR-code with private key for another wallet with some funds

Reproduction:
1. Scan QR-code from prerequisites
1. Withdraw screen opens - OK
1. Try to withdraw funds to your wallet
1. No mater what amount user wants to withdraw error is shown - WRONG - user should be able to withdraw funds

See attached screenshot with highlighted issues:
1. `xdai` instead of `pDai` or `sunDAI`
1. It shows "target" (user's) balance but that's not clear
1. Doesn't show balance of "source" account (the one from QR code)
1. Error message is shown no mater the amount user enters

Screenshot_20190413-160428_mod

### PR DESCRIPTION
**Pull request by TimDaub**
This PR is based on #45. It fixes #29 for the sundai branch.

It addresses the change proposals by @sunify in #33.

Tested:

- Withdrawing from Chrome desktop with metamask
- Withdrawing from Firefox desktop without metamask

### CODE REVIEW

**Comment by sunify** [comment]
Why not add this commit into #33 and cherry-pick commit from `sunDai` branch? Then merge it into master and rebase sunDai

**Comment by TimDaub** [comment]
> Why not add this commit into #33 and cherry-pick commit from `sunDai` branch? Then merge it into master and rebase sunDai

OK will try. Closing this.

**Comment by TimDaub** [comment]
> OK will try. Closing this.

Actually it's not as easy as `git cherry-pick 9d09791` as there are other variables added to `App.js` in other commits of branch sunDai that are expected by 9d09791.  Will hence simply reopen this PR and let go #33 go unmaintained for now.

File: src/components/WithdrawFromPrivate.js [hunk 1]
```diff
@@ -64,16 +66,18 @@ export default class SendToAddress extends React.Component {
...
         console.log("SWITCH TO LOADER VIEW...")
         this.props.changeView('loader')
         setTimeout(()=>{window.scrollTo(0,0)},60)
         //console.log("metaAccount",this.state.metaAccount,"amount",this.props.web3.utils.toWei(amount,'ether'))
 
-        const weiAmount = web3.utils.toWei(amount, "ether")
+        // NOTE: Amount needs to be cast to a string here.
+        const weiAmount = web3.utils.toWei(""+amount, "ether")
         const color = await xdaiweb3.getColor(pDaiTokenAddr);
+        console.log("color", color);
```
Comment (TimDaub) [inline src/components/WithdrawFromPrivate.js]:
```suggestion
```

File: src/App.js [hunk 2]
```diff
@@ -162,7 +162,8 @@ else if (window.location.hostname.indexOf("sundai.io") >= 0) {
   ERC20TOKEN = false;
   ERC20IMAGE = false;
 }
-else if (window.location.hostname.indexOf("sundai.local") >= 0) {
+else if (window.location.hostname.indexOf("sundai.local") >= 0 ||
+         window.location.hostname.indexOf("sundai.fritz.box") >= 0) {
```
Comment (TimDaub) [inline src/App.js]: This is for local network to debug. Shouldn't cause any issues.

File: src/components/WithdrawFromPrivate.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ import i18n from '../i18n';
 
 let pollInterval
 let metaReceiptTracker = {}
+const BN = Web3.utils.BN;
```
Comment (TimDaub) [inline src/components/WithdrawFromPrivate.js]:
```suggestion
```

**Pull request merged**